src/realtime/src/realtime/_async/channel.py

```diff
@@ -453,6 +473,11 @@ def on_presence_leave(
         :return: The Channel instance for method chaining.
         """
         self.presence.on_leave(callback)
+        if self.is_joined:
+            logger.info(
+                f"channel {self.topic} resubscribe due to change in presence callbacks on joined channel"
+            )
+            asyncio.create_task(self._resubscribe())
```

Comment:
@o-santi not sure if this `create_task` is appropriate here, should I hold a reference for the task to make sure it is finished/cancelled?

Reply:
Yes, `create_task` spawns it in the background, which is a problem because we won't know when its done. However, I'm not sure I understand the goal here. `on_presence_leave` is the time of registering the callback. Most of the time, the user won't even be subscribed when this is called. Did you mean to register a callback on the presence leave event for resubscribing?